### PR TITLE
Add Official Name page changes for Account

### PR DESCRIPTION
**Pull request by wilpow-github**
### Context

We’re building an account page where users can update both their ID and DQT details.

### Changes proposed in this pull request

Add official name change pages where User can submit request to update their DQT record

### Checklist

-   [x] Attach to Trello card
-   [ ] Rebased master
-   [ ] Cleaned commit history
-   [ ] Tested by running locally
-   [ ] Reminder created to manually clean any removed app settings post deployment

### CODE REVIEW

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Account/OfficialName/Details.cshtml.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        _linkGenerator = linkGenerator;
+        _dqtApiClient = dqtApiClient;
+    }
+
+    public ClientRedirectInfo? ClientRedirectInfo => HttpContext.GetClientRedirectInfo();
+
+    [BindProperty]
+    [Display(Name = "First name", Description = "Or given names")]
+    [Required(ErrorMessage = "Enter your first name")]
+    [StringLength(100, ErrorMessage = "First name must be 100 characters or less")]
```
Comment (wilpow-github) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Account/OfficialName/Details.cshtml.cs:28]: Assuming the 100 character limit is a DQT thing

File: dotnet-authserver/src/TeacherIdentity.AuthServer/DataAnnotationsExtensions.cs [hunk 2]
```diff
@@ -52,3 +52,47 @@ public IsPastDateAttribute(Type type)
...
+        return ValidationResult.Success!;
+    }
+}
+
+[AttributeUsage(AttributeTargets.Property | AttributeTargets.Field | AttributeTargets.Parameter)]
+public class FileExtensionsAttribute : ValidationAttribute
+{
+    private List<string> AllowedExtensions { get; set; }
+
+    public FileExtensionsAttribute(string fileExtensions)
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/DataAnnotationsExtensions.cs:82]:
```suggestion
    public FileExtensionsAttribute(params string[] fileExtensions)
```

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Account/OfficialName/RequireAuthenticationMilestoneAttribute.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
+using System.Security.Claims;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using TeacherIdentity.AuthServer.Services.DqtApi;
+
+namespace TeacherIdentity.AuthServer.Pages.Account.OfficialName;
+
+[AttributeUsage(AttributeTargets.Class, AllowMultiple = false)]
+public class OfficialNameChangeEnabledAttribute : Attribute, IAsyncPageFilter, IOrderedFilter
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Account/OfficialName/RequireAuthenticationMilestoneAttribute.cs:9]: (File name seems off here too)
```suggestion
public class CheckOfficialNameChangeIsEnabledAttribute : Attribute, IAsyncPageFilter, IOrderedFilter
```

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Account/OfficialName/Details.cshtml.cs [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+        if (NamesUnchanged())
+        {
+            ModelState.AddModelError(nameof(FirstName), "The name entered matches your official name");
+            return this.PageWithErrors();
+        }
+
+        return Redirect(_linkGenerator.AccountOfficialNameEvidence(FirstName!, MiddleName ?? String.Empty, LastName!, ClientRedirectInfo));
+    }
+
+    public override async Task OnPageHandlerExecutionAsync(PageHandlerExecutingContext context, PageHandlerExecutionDelegate next)
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Account/OfficialName/Details.cshtml.cs:67]: Can we not use the filter attribute here? Could stash the `TeacherInfo` object in `HttpContext.Items` in the filter if avoiding duplicate API call is the reason.

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Services/DqtApi/TeacherInfo.cs [hunk 5]
```diff
@@ -5,6 +5,8 @@ public record TeacherInfo
     public required string Trn { get; init; }
     public required string FirstName { get; init; }
     public required string LastName { get; init; }
+    // check this is right
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Services/DqtApi/TeacherInfo.cs:8]: It is right :-)

File: dotnet-authserver/src/TeacherIdentity.AuthServer/Services/DqtEvidence/IUserImportStorageService.cs [hunk 6]
```diff
@@ -0,0 +1,6 @@
+namespace TeacherIdentity.AuthServer.Services.UserImport;
```
Comment (gunndabad) [inline dotnet-authserver/src/TeacherIdentity.AuthServer/Services/DqtEvidence/IUserImportStorageService.cs:1]:
```suggestion
namespace TeacherIdentity.AuthServer.Services.DqtEvidence;
```

**Review by gunndabad — APPROVED** [review]

**Pull request merged**